codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/ManifestWriter.java

```diff
@@ -117,25 +140,64 @@ private static Node buildManifestFile(
         }
         generatedFiles = generatedFiles.stream().sorted().collect(Collectors.toList());
 
-        manifestNodes.put(StringNode.from("module"), StringNode.from(settings.getModuleName()));
-        minStandard.ifPresent(symbolDependency ->
-                manifestNodes.put(StringNode.from("go"), StringNode.from(symbolDependency.getVersion())));
+        manifestNodes.put(StringNode.from("module"), StringNode.from(moduleName));
+        minimumGoVersion.ifPresent(version -> manifestNodes.put(StringNode.from("go"),
+                StringNode.from(version)));
         manifestNodes.put(StringNode.from("dependencies"), ObjectNode.objectNode(dependencyNodes));
         manifestNodes.put(StringNode.from("files"), ArrayNode.fromStrings(generatedFiles));
-        manifestNodes.put(StringNode.from("unstable"),
-                BooleanNode.from(settings.getService(model).getTrait(UnstableTrait.class).isPresent()));
+        manifestNodes.put(StringNode.from("unstable"), BooleanNode.from(isUnstable));
 
         return ObjectNode.objectNode(manifestNodes).withDeepSortedKeys();
     }
 
     private static Map<String, String> gatherMinimumDependencies(
             Stream<SymbolDependency> symbolStream
     ) {
-        return SymbolDependency.gatherDependencies(symbolStream, GoDependency::mergeByMinimumVersionSelection)
-                .entrySet().stream()
-                .flatMap(entry -> entry.getValue().entrySet().stream())
-                .collect(Collectors.toMap(
-                        Map.Entry::getKey, entry -> entry.getValue().getVersion(), (a, b) -> b, TreeMap::new));
+        return SymbolDependency.gatherDependencies(symbolStream,
+                GoDependency::mergeByMinimumVersionSelection).entrySet().stream().flatMap(
+                entry -> entry.getValue().entrySet().stream()).collect(
+                Collectors.toMap(Map.Entry::getKey, entry -> entry.getValue().getVersion(), (a, b) -> b, TreeMap::new));
+    }
+
+    public static Builder builder() {
+        return new Builder();
     }
 
+    public static class Builder implements SmithyBuilder<ManifestWriter> {
+        private String moduleName;
+        private FileManifest fileManifest;
+        private List<SymbolDependency> dependencies;
```

Comment:
Using Smithy's `BuilderRef` for these types can be helpful. https://github.com/skmcgrail/smithy/blob/rules-engine/smithy-rules-engine/src/main/java/software/amazon/smithy/rulesengine/traits/EndpointTestsTrait.java#L93

Ensures the list is always copied from the builder, is initialized to an empty list on copy if it was never set etc.